### PR TITLE
feat(vouchers): POS receipt

### LINKED ISSUE #184: (feature request): Allow Voucher receipts to have custom titles
We use vouchers as our generic documents for recording new transactions for the Posting Journal.  For internal use, the term "voucher" is appropriate, however, vouchers has limited use to send to clients or give as receipts because "voucher" doesn't make a lot of sense to them.

I propose that we create a way to change the voucher title by attaching the voucher title to the transaction type.  That way, instead of a "Voucher", the patient will receive a "Credit Note" which is much clearer.

### LINKED ISSUE #185: (feature request): Implement a voucher POS receipt
The voucher module can be used in any location to record any type of transactions.  I propose that we create a POS receipt for the vouchers so that they might benefit from the thermal printers were appropriate.

### PR DESCRIPTION
**Pull request by jniles**
This commit implements a POS receipt for the voucher page.  This receipt
also puts the transaction type label (if available) at the top of the receipt and
 orders by debits/credits.  Credits are rendered as `(number)` to save space.

Fixes #184. Fixes #185.

voucherposreceipt
_Fig 1: The Voucher POS Receipt_

---
Thank you for contributing!

Before submitting this pull request, please verify that you have:
 - [x] Run your code through JSHint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
 - [x] Run integration tests.
 - [x] Run end-to-end tests.
 - [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

### CODE REVIEW

File: server/controllers/finance/vouchers.js [hunk 1]
```diff
@@ -22,11 +22,13 @@ const uuid = require('node-uuid');
 
 const util = require('../../lib/util');
 const db   = require('../../lib/db');
+const barcode = require('../../lib/barcode');
```
Comment (sfount) [inline server/controllers/finance/vouchers.js]: Is this import needed anymore?

File: server/lib/renderers/pdf.js [hunk 2]
```diff
@@ -47,12 +47,12 @@ exports.posReceiptOptions  = {
 
 // smaller format for providing identifications/ receipts with reduced information
 exports.reducedCardOptions = {
-  pageWidth : '75mm',
-  pageHeight : '125mm',
-  marginLeft : '5mm',
-  marginRight : '5mm',
-  marginTop : '5mm',
-  marginBottom : '5mm',
+  pageWidth : '75mm',       // 2.95in
```
Comment (sfount) [inline server/lib/renderers/pdf.js]: 👍

File: server/controllers/finance/vouchers.js (hunk 1 above)
Comment (jniles) [inline server/controllers/finance/vouchers.js]: Doesn't look like it.   Good catch!

File: server/lib/renderers/pdf.js (hunk 2 above)
Comment (jniles) [inline server/lib/renderers/pdf.js]: Hey man, when you don't have internet to look these things up and HP doesn't support the metric system ...

**Pull request merged**